File: crates/laya-parse/tests/fixtures/sample.php
<?php

namespace App\Store;

const VERSION = "1.0";

/** A key store. */
class KeyStore
{
    private array $map = [];

    public function get(string $key): string
    {
        return $this->map[$key] ?? "";
    }

    public function set(string $key, string $value): void
    {
        $this->map[$key] = $value;
    }
}

trait Loggable
{
    public function log(string $msg): void
    {
        echo $msg;
    }
}

function helper(int $x): int
{
    return $x * 2;
}

File: crates/laya-parse/tests/fixtures/sample.rs
//! Module docs.
use std::collections::HashMap;

/// A key-value store.
pub struct MoonStore {
    map: HashMap<String, String>,
}

pub const MAX_KEYS: usize = 1024;

pub trait Store {
    fn get(&self, key: &str) -> Option<String>;
}

impl Store for MoonStore {
    /// Fetch a key.
    fn get(&self, key: &str) -> Option<String> {
        let v = self.map.get(key);
        let v = v.cloned();
        if v.is_none() {
            return None;
        }
        let out = v;
        let out = out.map(|s| s.to_string());
        out
    }
}

fn helper() -> u32 {
    42
}
